scan/main.py

```diff
@@ -85,33 +96,97 @@ def _run_crew(crew):
     try:
         return crew.kickoff()
     except RateLimitError as e:
-        if e.code == "insufficient_quota":
-            raise RuntimeError(
-                "Insufficient quota. Please check your OpenAI plan and billing details."
-            ) from e
+        logger.error(f"Whoops! We hit a rate limit: {e}")
+        raise
+    except Exception as e:
+        logger.error(f"Oh no! Something unexpected happened: {e}")
+        raise
+
+def classify_input(input_text):
+    try:
+        response = client.chat.completions.create(
+            model="gpt-4",
+            messages=[
+                {"role": "system", "content": "You are an AI assistant. Classify the following input as either 'PFC task' or 'General task' based on its content. For PFC tasks, provide a brief description of which PFC functions might be involved (e.g., decision-making, emotional regulation, executive functions)."},
+                {"role": "user", "content": f"Classify this input: {input_text}"}
+            ],
+            max_tokens=100,
+            temperature=0.3
+        )
+        if response.choices and response.choices[0].message and response.choices[0].message.content:
+            classification = response.choices[0].message.content.strip()
+            return classification
         else:
-            raise
+            logger.error("Received an unexpected response format from OpenAI API.")
+            return "General task"
+    except Exception as e:
+        logger.error(f"Error in classifying input: {str(e)}")
```

Comment:
f-strings can have unexpected results

```suggestion
        logger.error("Error in classifying input: %s", e)